src/data/iterative_dmatrix.cc

```diff
@@ -63,6 +63,13 @@ void GetCutsFromRef(std::shared_ptr<DMatrix> ref_, bst_feature_t n_features, Bat
     }
   };
   auto ellpack = [&]() {
+    // workaround ellpack being initialized from CPU.
+    if (p.gpu_id == Context::kCpuId) {
```

Comment:
I don't understand what is happening here, and it looks weird with two statements having the same condition.

Reply:
If the device is on CPU, we try the device ordinal in reference dmatrix. After the assignment from the reference dmatrix, if the device is still on cpu, we just use 0 as a workaround.

This happens when both the current qdm and the reference dmatrix are initialized on CPU but we train the booster on GPU.

I'm working to unify the device ordinal, we will have a single context from booster instead. This PR is quick fix for backport.